File: week18/p_42885.py
from collections import deque
people = [30,30,30]		
limit = 100

def solution(people, limit):
    people.sort()
    people = deque(people)
    answer = 0
    while people:
        left = people.popleft()
        
        if not people:
            return answer + 1

        else:
            right = people.pop()
            if right + left > limit: #작은 경우 생각 안해도 되는게, limit 보다 클때만 보트를 하나씩 더해주면됨. 
                people.appendleft(left)
        answer += 1 #어차피 1명밖에 못탐. 
    return answer

File: week20/p_42885.js

// 효율성 0
// function solution(people, limit) {
//     var answer = 0;    
//     people.sort((a,b)=> (b-a))
    
//     while(people.length){
//         let right = people.shift()
//         if(people.length===0){
//             answer += 1
//             return answer
//         }
//         else {
//             let left = people.pop()
//             if(right + left > limit){
//                 people.push(left)   
//             }

//             answer += 1
//         }
//     }
//     return answer;
// }